client/src/navigator/Drawer/Drawer.js

```diff
@@ -128,100 +132,188 @@ const generateTabs = (tabData) => tabData.map((element, index) => (
+            color={tabColors[1]}
+            style={{ position: 'absolute' }}
+          />
+        </View>
+      ),
+    })}
+  />
+))
+
+const generateLearnerTabs = (tabData) => tabData.map((element, index) => (
+  <Drawer.Screen
+    key={element._id}
+    name={element._id}
+    component={TabNavigator}
+    options={() => ({
+      drawerLabel: () => (
+        <View style={tabStyles.container}>
+          <View>
+            <Text
+              style={tabStyles.title}
+              fontFamily="heading"
+              fontWeight="regular"
+              fontStyle="normal"
+            >
+              {element.name}
+            </Text>
+            <Text style={tabStyles.units}>
+              {generateUnitLabel(element.num_units)}
+            </Text>
+          </View>
+        </View>
+      ),
+      drawerIcon: () => (
+        <View>
+          <NumberBox number={index + 1} learner noMargin />
+        </View>
       ),
     })}
   />
 ))
 
 const DrawerMenuContainer = (props) => {
-  const { state, ...rest } = props
+  const {
+    state, firstRouteNames, secondRouteNames, ...rest
+  } = props
   const newState = { ...state }
-  const drawerApply = true
+
+  const middleChildComponent = (
+    <>
+      <View style={drawerStyles.container}>
+        <Pressable
+          style={drawerStyles.pressable}
+          forceInset={{
+            top: 'always',
+            horizontal: 'never',
+          }}
+        >
+          <Text
+            fontWeight="regular"
+            color="gray.dark"
+            fontSize="sm"
+            textAlign="left"
+          >
+            {`${i18n.t('dialogue.learnIndigenousLanguage')} `}
+            <Text fontFamily="heading" fontWeight="regular" fontStyle="normal">
+              {i18n.t('actions.startLearning')}
+            </Text>
+          </Text>
+          <StyledButton
+            title={i18n.t('actions.searchCourses')}
+            fontSize="sm"
+            variant="learner_primary"
+            onPress={() => props.navigation.navigate('Apply', { from: 'HomeBaseCase' })}
+          />
+        </Pressable>
+      </View>
+
+      <StyledButton
+        title="CONTRIBUTOR"
+        fontSize={15}
+        variant="contributor"
+        onPress={() => props.navigation.navigate('Apply', { from: 'HomeBaseCase' })}
+      />
+    </>
+  )
 
   return (
     <>
       <DrawerContentScrollView {...props}>
         <DrawerMenu {...props} />
-        <DrawerItemList state={newState} {...rest} />
 
-        {drawerApply ? (
-          <View style={drawerStyles.container}>
-            <Pressable
-              style={drawerStyles.pressable}
-              forceInset={{
-                top: 'always',
-                horizontal: 'never',
-              }}
+        <StyledButton
+          title="LEARNER"
+          fontSize={15}
+          variant="learner"
+          onPress={() => props.navigation.navigate('Apply', { from: 'HomeBaseCase' })}
+        />
+
+        <SplitDrawerItemList
```

Comment:
got a console log warning that one of the props is an object instead of an array